org.eclipse.lyo.oslc4j.provider.jena/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/JenaModelHelper.java

```diff
@@ -489,16 +492,38 @@ public static void skolemize(final Model m, Function<BlankNodeId, String> skolem
 		}
 	}
 
+        private static Map<String, Class> getRdfTypeClasses(Package pkg) {
```

Comment:
I think it would be good to refactor the method that checks if the class is an OSLC class. Maybe it will be my aciton point. I would also like to add some heuristics to make the OSLC resource class declaration by hand less verbose.

Reply:
I neither like the idea of scanning types on every unmarshal, registering packages may help in that.